core/src/main/java/kafka/server/share/ShareFetchUtils.java

```diff
@@ -128,4 +131,13 @@ static long offsetForEarliestTimestamp(TopicIdPartition topicIdPartition, Replic
                 Optional.empty(), true).timestampAndOffsetOpt();
         return timestampAndOffset.isEmpty() ? (long) 0 : timestampAndOffset.get().offset;
     }
+
+    static int leaderEpoch(ReplicaManager replicaManager, TopicPartition tp) {
```

Comment:
Returning an OptionalInt would be a bit nicer than throwing there (maybe). If we actually want a helper method that throws, we should incorporate that into the name (e.g., `leaderEpochOrThrow`)

Reply:
With an optionalInt again we need to check and pass the right error. So either we have this method or create `leaderEpochOrThrow` in replica manager.